lib/core/src/Cardano/Wallet/Api/Link.hs

```diff
@@ -0,0 +1,520 @@
+        ( HasType (ApiT WalletId) w
+        , style ~ 'Shelley
+        )
+    => w
+    -> (Method, Text)
+createTransaction w =
+    endpoint @(Api.CreateTransaction Net) (wid &)
+  where
+    wid = w ^. typed @(ApiT WalletId)
+
+listTransactions
+    :: forall (style :: WalletStyle) w.
+        ( HasType (ApiT WalletId) w
+        , KnownSymbol (WalletStyleSymbol style)
+        )
+    => w
+    -> Maybe Iso8601Time
+    -> Maybe Iso8601Time
+    -> Maybe SortOrder
+    -> (Method, Text)
+listTransactions w inf sup order = discriminate @style
+    (endpoint @(Api.ListTransactions Net) mkURL)
+    (endpoint @(Api.ListByronTransactions Net) mkURL)
+  where
+    wid = w ^. typed @(ApiT WalletId)
+    mkURL mk = mk wid inf sup (ApiT <$> order)
+
+postTransactionFee
+    :: forall (style :: WalletStyle) w.
+        ( HasType (ApiT WalletId) w
+        , style ~ 'Shelley
+        )
+    => w
+    -> (Method, Text)
+postTransactionFee w =
+    endpoint @(Api.PostTransactionFee Net) (wid &)
+  where
+    wid = w ^. typed @(ApiT WalletId)
+
+deleteTransaction
+    :: forall (style :: WalletStyle) w t.
+        ( HasType (ApiT WalletId) w
+        , HasType (ApiT (Hash "Tx")) t
+        , KnownSymbol (WalletStyleSymbol style)
+        )
+    => w
+    -> t
+    -> (Method, Text)
+deleteTransaction w t = discriminate @style
+    (endpoint @Api.DeleteTransaction mkURL)
+    (endpoint @Api.DeleteByronTransaction mkURL)
+  where
+    wid = w ^. typed @(ApiT WalletId)
+    tid = ApiTxId (t ^. typed @(ApiT (Hash "Tx")))
+    mkURL mk = mk wid tid
+
+--
+-- Stake Pools
+--
+
+listStakePools
+    :: (Method, Text)
+listStakePools =
+    endpoint @Api.ListStakePools id
+
+joinStakePool
+    :: forall (style :: WalletStyle) s w.
+        ( HasType (ApiT PoolId) s
+        , HasType (ApiT WalletId) w
+        , style ~ 'Shelley
+        )
+    => s
+    -> w
+    -> (Method, Text)
+joinStakePool s w =
+    endpoint @(Api.JoinStakePool Net) (\mk -> mk sid wid)
+  where
+    sid = s ^. typed @(ApiT PoolId)
+    wid = w ^. typed @(ApiT WalletId)
+
+quitStakePool
+    :: forall (style :: WalletStyle) s w.
+        ( HasType (ApiT PoolId) s
+        , HasType (ApiT WalletId) w
+        , style ~ 'Shelley
+        )
+    => s
+    -> w
+    -> (Method, Text)
+quitStakePool s w =
+    endpoint @(Api.QuitStakePool Net) (\mk -> mk sid wid)
+  where
+    sid = s ^. typed @(ApiT PoolId)
+    wid = w ^. typed @(ApiT WalletId)
+
+delegationFee
+    :: forall (style :: WalletStyle) w.
+        ( HasType (ApiT WalletId) w
+        , style ~ 'Shelley
+        )
+    => w
+    -> (Method, Text)
+delegationFee w =
+    endpoint @Api.DelegationFee (wid &)
+  where
+    wid = w ^. typed @(ApiT WalletId)
+
+--
+-- Network Information
+--
+
+getNetworkInfo
+    :: (Method, Text)
+getNetworkInfo =
+    endpoint @Api.Network id
+
+--
+-- Proxy
+--
+
+postExternalTransaction
+    :: (Method, Text)
+postExternalTransaction =
+    endpoint @Api.PostExternalTransaction id
+
+
```

Comment:
We should probably delete the second empty line here. ^^

(https://github.com/input-output-hk/cardano-wallet/wiki/Coding-Standards#use-only-a-single-blank-line-between-top-level-definitions)